Icinga Kubernetes Role and Icinga Kubernetes Web install

gianmarco-mameli opened the pull request:
Hello,
in the past month I tried the new Icinga Kubernetes daemon and module and, because my Icinga Infrastructure is outside of my K3S Cluster, I preferred to install the packages on the master nodes.
Using most of the excellent existent work, I copied the structures and implemented the installation of the daemon and db schema, in addition to the Icingaweb2 dedicated module.
The tasks are working and running correctly on my Icinga test installation (both existent and from scratch), so if you're interested I appreciate to check and merge my work.
I think I have covered most of the docs and files that needed update, but I certainly not a good expert to add the Molecule tests needed, so in that case maybe I need some help, but I'm already trying to fill the gaps

Thanks

PS. In my spare time I'm also implementing the same work for Dependencies and Fileshipper modules.

mkayontour commented:
Hi, at the first glance this looks really good, thank you for the work. We'll review it as soon as possible and give you feedback.

Donien commented on roles/kubernetes/defaults/main.yml:
```diff
@@ -0,0 +1,19 @@
+kubernetes_database_schema: "/usr/share/icinga-kubernetes/schema/{{ kubernetes_database_type }}/schema.sql"
+kubernetes_packages:
+  - icinga-kubernetes
+kubernetes_kubeconfig_path: "{{ ansible_env.HOME }}/.kube/config"
```
```suggestion
kubernetes_kubeconfig_path: "{{ kubernetes_config_dir }}/kube_config"
```

Donien reviewed (changes requested):
Looks good to me so far.

The only thing is that I'm running into a bug where my apache web server fails completely for some reason a few minutes after having k3s, Icinga Web + Icinga-Kubernetes running (service okay, no errors in log, but 404 for every url).  
This is likely unrelated though.

gianmarco-mameli commented on doc/role-kubernetes/role-kubernetes.md:
```diff
@@ -0,0 +1,75 @@
+| `kubernetes_packages` | `List` | Defines the packages to install for Icinga Kubernetes. | `[icinga-kubernetes]` |
+| `kubernetes_service_name` | `String` | Defines the name of the Icinga Kubernetes service. | `icinga-kubernetes` |
+| `kubernetes_user` | `String` | Defines the user for the Icinga Kubernetes service. | `icinga-kubernetes` |
+| `kubernetes_kubeconfig_path` | `String` | Defines the path for the kubeconfig file, if not in standard path. | `{{ ansible_env.HOME }}/.kube/config` |
```
you're right, maybe `ansible_username`, I did not find anything on the module documentation for a default dir, usually a k3s or k8s installation leaves the kubeconfig file under /<user>/.kube.
I'm gonna fix that path as default `/etc/icinga-kubernetes/kube_config` and leaving the user change it on his side
Is that ok to you?

gianmarco-mameli commented:
ok, I made the fix you suggested, thanks

Donien reviewed (approved):
We need a changelog fragment. 

changelogs/fragments/feature_kubernetes.yml:

```yaml
major_changes:
  - Add a role for the installation and configuration of `Icinga for Kubernetes <https://icinga.com/docs/icinga-for-kubernetes/latest/>`_.
  - Add tasks to role :code:`icingaweb2` to install and configure `Icinga for Kubernetes Web <https://icinga.com/docs/icinga-kubernetes-web/latest/doc/02-Installation/>`_.
```

@mkayontour, do you want to have another look at this as well?

gianmarco-mameli commented:
changelog fragment ok

mkayontour commented on roles/kubernetes/templates/kubernetes.ini.j2:
The filename is wrong, kubernetes.yml.j2 should be the name. It is not a ini file config.

mkayontour commented on roles/kubernetes/tasks/manage_service.yml:
Why is the override configuration created and doesn't override anything? 

I don't want to create files if they are not needed. And the override configuration cannot be edited without changing the "source". 

It should be a external file which can be defined in your own Ansible directory and if available will be deployed.

gianmarco-mameli replied:
I see the problem, let me correct the file and make some tests. THe file must be available in the collection to permit a custom kubeconfig file. I get back to you after the tests, thanks

mkayontour commented on roles/kubernetes/templates/icinga-kubernetes-override.conf.j2:
This file is not a template and should be a optional static file which is available in your own ansible directory.

mkayontour reviewed (changes requested):
Overall I'm good with the general code, I'm having a problem with the variable names which may overlap with other [kubernetes roles](https://github.com/geerlingguy/ansible-role-kubernetes) and helps with visibility on your ansible code. 
We should have a prefix like "icinga_" 

And the override file should not be created as it is right now, see the comments on the file.

gianmarco-mameli commented:
Hi @mkayontour I've updated the role with your suggestions, I also modified the application of the service override only if is defined a custom path for the kubeconfig.
The name of the role is now `icinga_kubernetes` to permit adding the prefix on every variable and compliance with the linting rule of variable prefix name.
I think also doc is updated correctly, let me know

Thanks

gianmarco-mameli commented:
Hello, do you have any news for this PR? I have implemented other modules installation, but I want to be sure that this is ok so I can replicate the same procedure, thanks

mkayontour reviewed (approved):
Excuse the late response, this is really good stuff. 

Right now, we are deprecating the collection in this space. The collection will move to github.com/netways. 

We will tag a working bugfix release tomorrow and then push a release with new features on the NETWAYS Repo. This will happen in a few days, so it would be good if you then can create the PR against the new repo and it will be in the next release.

gianmarco-mameli commented:
No problem, that's good, I'll be waiting for the new repo to open the PR

Have a nice day

gianmarco-mameli commented:
Hi all, I moved back this PR as Draft because I found that the icinga-kubernetes package change the mode of managemet of the daemon that retrieve the data. I'm already working on it and I will submit the mods and revert the PR to ready when it's finished, thanks

Donien reviewed (changes requested):
Just a few FQCN namespace changes.

I'll merge afterwards :)

Thanks again for your efforts with this!

gianmarco-mameli commented:
Hi @Donien, thank you, sorry I was out of home for some days of vacancy
